apps/comps/components/agents-table/index.tsx

```diff
@@ -76,9 +76,9 @@ export const AgentsTable: React.FC<AgentsTableProps> = ({
   const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({
     yourShare: session.ready && session.isAuthenticated,
   });
-  const [selectedAgent, setSelectedAgent] = useState<AgentCompetition | null>(
-    null,
-  );
+  const [selectedAgent, setSelectedAgent] = useState<
+    RouterOutputs["competitions"]["getAgents"]["agents"][number] | null
```

Comment:
I think at some point it might be nice to give these types name aliases so things are a bit easier to read, but not important right now.  Once everything in the frontend is using this method of inferring the types from the backend, we can delete all the existing types defined in the frontend's types file and add aliases for these RouterOutputs-derived types there instead as a nice readability cleanup